### PR TITLE
Update login and controller to use standard image setting format

### PR DESCRIPTION
**Pull request by heyealex**
The login and controller for slurm-gcp v5 had been using the format from Slurm on GCP, and therefore have been inconsistent with partitions/node-groups. This commit fixes that by hard deprecating the old format and adding the new "instance_image" format.

This PR should fix issue #753 with the ubuntu slurm-gcp v5 example blueprint.

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/compute/schedmd-slurm-gcp-v5-node-group/variables.tf [hunk 1]
```diff
@@ -104,6 +108,24 @@ variable "instance_image" {
   }
 }
 
+variable "source_image_project" {
+  type        = string
+  description = "Alternate method to instance_image for specifying a project hosting the custom VM image."
```
Comment (cboneti) [inline community/modules/compute/schedmd-slurm-gcp-v5-node-group/variables.tf:113]: Nit: here I would explicitly say: "We recommend using 'instance_image' instead".

File: community/modules/scheduler/schedmd-slurm-gcp-v5-controller/variables.tf [hunk 2]
```diff
@@ -483,37 +483,57 @@ variable "slurm_conf_tpl" {
...
+  }
+  validation {
+    condition     = length(var.instance_image) == 0 || can(var.instance_image["family"]) != can(var.instance_image["name"])
+    error_message = "Exactly one of \"family\" and \"name\" must be provided in var.instance_image."
+  }
+}
+
+variable "source_image_project" {
+  type        = string
+  description = "Alternate method to instance_image for specifying a project hosting the custom VM image."
```
Comment (cboneti) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-controller/variables.tf:523]: same nit here.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 3]
```diff
@@ -247,37 +247,57 @@ variable "startup_script" {
...
+    can(var.instance_image["family"]) || can(var.instance_image["name"])) == can(var.instance_image["project"])
+    error_message = "The \"project\" is required if \"family\" or \"name\" are provided in var.instance_image."
+  }
+  validation {
+    condition     = length(var.instance_image) == 0 || can(var.instance_image["family"]) != can(var.instance_image["name"])
+    error_message = "Exactly one of \"family\" and \"name\" must be provided in var.instance_image."
+  }
+}
+
+variable "source_image_project" {
```
Comment (cboneti) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:285]: and here.

**Review by cboneti — APPROVED** [review]
approved with a nit, feel free to change and submit, or submit without the change.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-controller/variables.tf (hunk 2 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-controller/variables.tf]: Done

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf (hunk 3 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:285]: Done

**Pull request merged**